src/remoted/manager.c

```diff
@@ -1247,9 +1279,8 @@ void *update_shared_files(__attribute__((unused)) void *none) {
 
 void free_pending_data(pending_data_t *data) {
     if (!data) return;
-    if (data->message) free(data->message);
-    if (data->keep_alive) free(data->keep_alive);
-    free(data);
+    os_free(data->message);
```

Comment:
Why set `data->message` and `data` to `NULL`, if we're losing them?

Reply:
This is the method used to free the memory allocated for the entries in the `pending_data` hash table. If an entry is removed, we have to release the memory of either the message as well as the entry itself.